packages/beacon-node/src/chain/chain.ts

```diff
@@ -254,9 +254,13 @@ export class BeaconChain implements IBeaconChain {
             pubkey2index: new PubkeyIndexMap(),
             index2pubkey: [],
           });
-    this.shufflingCache.processState(cachedState, cachedState.epochCtx.previousShuffling.epoch);
-    this.shufflingCache.processState(cachedState, cachedState.epochCtx.currentShuffling.epoch);
-    this.shufflingCache.processState(cachedState, cachedState.epochCtx.nextShuffling.epoch);
+
+    cachedState.epochCtx.shufflingCache = this.shufflingCache;
+    this.shufflingCache.set(cachedState.epochCtx.previousShuffling, cachedState.epochCtx.previousDecisionRoot);
+    this.shufflingCache.set(cachedState.epochCtx.currentShuffling, cachedState.epochCtx.currentDecisionRoot);
+    if (cachedState.epochCtx.nextShuffling !== null) {
+      this.shufflingCache.set(cachedState.epochCtx.nextShuffling, cachedState.epochCtx.nextDecisionRoot);
```

Comment:
the only time I see we use this `set` api is right after the constructor. Consider having these in `ShufflingCache` constructor instead and drop the `set` api

Reply:
Good idea!  Updated.